src/main/java/net/neoforged/neoforge/client/util/TooltipUtil.java

```diff
@@ -0,0 +1,171 @@
+                double base = baseModif.base.amount() + entityBase;
+                final double rawBase = base;
+                double amt = base;
+                double baseBonus = attr.value().getBonusBaseValue(stack);
+                for (AttributeModifier modif : baseModif.children) {
+                    if (modif.operation() == Operation.ADD_VALUE) base = amt = amt + modif.amount();
+                    else if (modif.operation() == Operation.ADD_MULTIPLIED_BASE) amt += modif.amount() * base;
+                    else amt *= 1 + modif.amount();
+                }
+                amt += baseBonus;
+                boolean isMerged = !baseModif.children.isEmpty() || baseBonus != 0;
+                MutableComponent text = attr.value().toBaseComponent(amt, entityBase, isMerged, flag);
+                tooltip.accept(padded(" ", text).withStyle(isMerged ? ChatFormatting.GOLD : ChatFormatting.DARK_GREEN));
+                if (Screen.hasShiftDown() && isMerged) {
+                    // Display the raw base value, and then all children modifiers.
+                    text = attr.value().toBaseComponent(rawBase, entityBase, false, flag);
+                    tooltip.accept(list().append(text.withStyle(ChatFormatting.DARK_GREEN)));
+                    for (AttributeModifier modifier : baseModif.children) {
+                        tooltip.accept(list().append(attr.value().toComponent(modifier, flag)));
+                    }
+                    if (baseBonus > 0) {
+                        attr.value().addBonusTooltips(stack, tooltip, flag);
+                    }
+                }
+            }
+
+            for (Holder<Attribute> attr : modifierMap.keySet()) {
+                if (baseModifs.containsKey(attr)) continue;
+                Collection<AttributeModifier> modifs = modifierMap.get(attr);
+                // Initiate merged-tooltip logic if we have more than one modifier for a given attribute.
+                if (modifs.size() > 1) {
+                    double[] sums = new double[3];
+                    boolean[] merged = new boolean[3];
+                    Map<Operation, List<AttributeModifier>> shiftExpands = new HashMap<>();
+                    for (AttributeModifier modifier : modifs) {
+                        if (modifier.amount() == 0) continue;
+                        if (sums[modifier.operation().ordinal()] != 0) merged[modifier.operation().ordinal()] = true;
+                        sums[modifier.operation().ordinal()] += modifier.amount();
+                        shiftExpands.computeIfAbsent(modifier.operation(), k -> new LinkedList<>()).add(modifier);
+                    }
+                    for (Operation op : Operation.values()) {
+                        int i = op.ordinal();
+                        if (sums[i] == 0) continue;
+                        if (merged[i]) {
+                            TextColor color = sums[i] < 0 ? TextColor.fromRgb(0xF93131) : TextColor.fromRgb(0x7A7AF9);
```

Comment:
Maybe utilize `Attribute::getStyle`?

Reply:
This code does need to be updated to account for `Attribute.Sentiment`, but cannot utilize `Attribute.getStyle` - this color is a separate color used for the merged tooltip.

Reply:
Resolved by adding `IAttributeExtension#getMergedStyle`